src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -792,5 +792,14 @@ size_t FileBasedWal::accessAllBuffers(std::function<bool(BufferPtr buffer)> fn)
     return count;
 }
 
+TermID FileBasedWal::getLogTerm(LogID id) {
+    TermID term = -1;
+    auto walIter = std::make_unique<FileBasedWalIterator>(shared_from_this(), id, id);
```

Comment:
You could try to read from memory cache first, it could be quite slow.

Reply:
I will modify it later

Reply:
done

Reply:
👌  LGTM